### PR TITLE
Recommendation for optios to registrations

### PR DESCRIPTION
**Pull request by generik0**
Hi @fir3pho3nixx 
Please consider these options.
I see great value in them.

I will add comments etc, if you approve the changes

### CODE REVIEW

File: src/Castle.Facilities.AspNetCore.Tests/Builders/WindsorContainerBuilder.cs [hunk 1]
```diff
@@ -33,12 +33,17 @@ public static IWindsorContainer New(IServiceCollection serviceCollection)
 		private static IWindsorContainer BuildWindsorContainer(IServiceCollection serviceCollection)
 		{
 			var container = new WindsorContainer();
-			serviceCollection.AddCastleWindsor(container, opts => opts.UseEntryAssembly(typeof(AspNetCoreFacilityTestCase).Assembly));
+			serviceCollection.AddCastleWindsor(container, opts =>
+			{
+				opts.UseEntryAssembly(typeof(AspNetCoreFacilityTestCase).Assembly);
+				opts.RegisterAndCrossWire<IAnyCrossWireAndRegisterService, AnyCrossWireAndRegisterService>();
```
Comment (fir3pho3nixx) [inline src/Castle.Facilities.AspNetCore.Tests/Builders/WindsorContainerBuilder.cs]: I like this idea. Let me bake it in.

File: src/Castle.Facilities.AspNetCore.Tests/Builders/WindsorContainerBuilder.cs [hunk 2]
```diff
@@ -33,12 +33,17 @@ public static IWindsorContainer New(IServiceCollection serviceCollection)
...
+				opts.RegisterHttpContextAccessor();
```
Comment (fir3pho3nixx) [inline src/Castle.Facilities.AspNetCore.Tests/Builders/WindsorContainerBuilder.cs]: Do we need this?

File: src/Castle.Facilities.AspNetCore.Tests/Builders/WindsorContainerBuilder.cs (hunk 1 above)
Comment (fir3pho3nixx) [reply src/Castle.Facilities.AspNetCore.Tests/Builders/WindsorContainerBuilder.cs]: Please have a look at this: https://github.com/fir3pho3nixx/Windsor/blob/aspnet-core-windsor-final/src/Castle.Facilities.AspNetCore/CastleWindsorRegistrationExtensions.cs#L133

File: src/Castle.Facilities.AspNetCore.Tests/Builders/WindsorContainerBuilder.cs (hunk 2 above)
Comment (fir3pho3nixx) [reply src/Castle.Facilities.AspNetCore.Tests/Builders/WindsorContainerBuilder.cs]: We can now do it like this:

```csharp
serviceCollection.AddCastleWindsor(container, opts =>
{
	opts.UseEntryAssembly(typeof(AspNetCoreFacilityTestCase).Assembly);
	opts.RegisterCrossWired<IAnyCrossWireAndRegisterService, AnyCrossWireAndRegisterService>();
	opts.Register<IHttpContextAccessor, HttpContextAccessor>();
});
```

**Comment by generik0** [comment]
@fir3pho3nixx I like the AddHttpContextAccessor(). In the summary we can add the url you sent me about MS not recommending it’s usagage.
I think it is a good feature as this is a very common option.
I know it can be done with the register,but this is unique feature of aspnetcore and I think should be in the options.

**Comment by fir3pho3nixx** [comment]
@generik0 - You twist my arm. 

How do you want this method to work? 

https://github.com/castleproject/Windsor/pull/389/commits/9a7328df261cb66a2e3c13293b80d7cdd2dc9c93#diff-bfc97737e500590c8704fa94c7ac2776R36

Here you only had it registered in the service collection as a singleton, should we cross wire it?

**Comment by generik0** [comment]
@fir3pho3nixx previously the IHttpContextAccessor apears to resolve both for container and service collection when it was added to the service collection. 
Is this not the best implementation? Default/standard standard
I think MS have made this implementation to be a little magical.

I.e. if it works without cross wiring, should we not just use this?

**Comment by fir3pho3nixx** [comment]
All added here: https://github.com/fir3pho3nixx/Windsor/blob/aspnet-core-windsor-final/src/Castle.Facilities.AspNetCore/CastleWindsorRegistrationExtensions.cs#L205

**Pull request closed without merging**